pkg/endpoint/bpf.go

```diff
@@ -614,14 +614,17 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, boo
 	os.RemoveAll(e.IPv4IngressMapPathLocked())
 
 	e.Mutex.Unlock()
-
+	e.getLogger().WithField(
```

Comment:
`getLogger` needs e.Mutex locked. :( sorry. you can logger := e.getLogger() before the unlock on line 616, then reuse it in the other places.

The lock is needed because getLogger may regenerate the fields reported in the logger and those need to be protected.